packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -371,15 +375,43 @@ export abstract class ElementHandle<
    * elements matching the given selector.
    */
   @throwIfDisposed()
-  @ElementHandle.bindIsolatedHandle
   async $$<Selector extends string>(
-    selector: Selector
+    selector: Selector,
+    options?: QueryOptions
   ): Promise<Array<ElementHandle<NodeFor<Selector>>>> {
-    const {updatedSelector, QueryHandler} =
-      getQueryHandlerAndSelector(selector);
-    return await (AsyncIterableUtil.collect(
-      QueryHandler.queryAll(this, updatedSelector)
-    ) as Promise<Array<ElementHandle<NodeFor<Selector>>>>);
+    if (options?.isolate === false) {
+      const {updatedSelector, QueryHandler} =
+        getQueryHandlerAndSelector(selector);
+      return await (AsyncIterableUtil.collect(
+        QueryHandler.queryAll(this, updatedSelector)
+      ) as Promise<Array<ElementHandle<NodeFor<Selector>>>>);
+    }
+    // If the handle is already isolated, then we don't need to adopt it
```

Comment:
WDYT about have a private method and attaching the decorator to it then calling it here?

Reply:
didn't get the idea with the private method but let's reuse the decorator